main.lua

```diff
@@ -177,3 +183,26 @@ end
 function calcDist(x1, y1, x2, y2)
 	return hypot(x2 - x1, y2 - y1)
 end
+
+function isWithinBox(x, y, x1, y1, width, height)
+	return x >= x1 and y >= y1 and x < (x1 + width) and y < (y1 + height)
+end
+
+function checkAndClickButton(x, y, button)
+	if isWithinBox(x, y, button.x, button.y, button.width, button.height) then
+		button.onPress()
+		return true
+	end
+	return false
+end
+
+function drawButton(button)
+	if isWithinBox(relMouse.x, relMouse.y, button.x, button.y, button.width, button.height) then
+		love.graphics.setColor(1, 1, 1, 0.3)
+		love.graphics.rectangle("fill", button.x, button.y, button.width, button.height)
+	end
```

Comment:
Would be nice to add an else here to pick a different color, and move the rectangle drawing outside so it unconditionally happens. That way, all buttons have a background no matter what, and when you hover, a highlight will happen. This can be for later